AutoCollection/HttpDependencies.ts

```diff
@@ -122,31 +122,40 @@ class AutoCollectHttpDependencies {
         // methods exist before invoking them.
         if (Util.canIncludeCorrelationHeader(client, requestParser.getUrl()) && telemetry.request.getHeader && telemetry.request.setHeader) {
             if (client.config && client.config.correlationId) {
-                const correlationHeader = telemetry.request.getHeader(RequestResponseHeaders.requestContextHeader);
-                if (correlationHeader) {
-                    const components = correlationHeader.split(",");
-                    const key = `${RequestResponseHeaders.requestContextSourceKey}=`;
-                    if (!components.some((value) => value.substring(0,key.length) === key)) {
-                        telemetry.request.setHeader(
-                            RequestResponseHeaders.requestContextHeader,
-                            `${correlationHeader},${RequestResponseHeaders.requestContextSourceKey}=${client.config.correlationId}`);
+                // getHeader returns "any" type in newer versions of node. In basic scenarios, this will be <string | string[] | number>, but could be modified to anything else via middleware
+                const correlationHeader = <any>telemetry.request.getHeader(RequestResponseHeaders.requestContextHeader)
+                let header: string; // cast correlationHeader to string
+                if (typeof correlationHeader === "string") {
+                    header = correlationHeader;
+                } else if (correlationHeader instanceof Array) { // string[]
+                    header = correlationHeader.join(",");
+                } else if (correlationHeader && typeof (correlationHeader as any).toString === "function") {
+                    // best effort attempt: requires well-defined toString
+                    try {
+                        header = (correlationHeader as any).toString();
+                    } catch (err) {
+                        Logging.warn("Outgoing request-context header could not be parsed. Correlation of requests may be lost.", err, correlationHeader);
```

Comment:
Do you think `parsed` might give users the false idea that we could read it but they sent headers in the wrong format?

Reply:
I guess I am using parsed as read right now, but it does introduce ambiguity. "Read" is more correct as to why correlation will not be working